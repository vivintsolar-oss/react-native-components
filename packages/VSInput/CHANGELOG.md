# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.2.8](https://github.com/vivintsolar-oss/react-native-components/compare/@vivintsolar-oss/native-vs-input@1.2.7...@vivintsolar-oss/native-vs-input@1.2.8) (2018-11-29)

**Note:** Version bump only for package @vivintsolar-oss/native-vs-input





## [1.2.8](https://github.com/vivintsolar-oss/react-native-components/compare/@vivintsolar-oss/native-vs-input@1.2.7...@vivintsolar-oss/native-vs-input@1.2.8) (2018-11-29)

**Note:** Version bump only for package @vivintsolar-oss/native-vs-input
