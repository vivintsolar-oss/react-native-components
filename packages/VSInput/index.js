import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text, TextInput, View, TouchableWithoutFeedback } from 'react-native';
import VSSvg from '@vivintsolar-oss/native-vs-icons';
import Colors from '@vivintsolar-oss/native-vs-constants/Colors';
import Touchable from '@vivintsolar-oss/native-vs-touchable';

export default class VSInput extends Component {
  constructor(props) {
    super(props);
    this.togglePw = this.togglePw.bind(this);
    this.state = {
      showPw: false,
    };
  }

  togglePw() {
    return this.setState({ showPw: !this.state.showPw });
  }

  borderState() {
    if (this.props.error) {
      return Colors.ERROR;
    }

    return Colors.BORDER_GRAY;
  }

  labelState() {
    if (this.props.error) {
      return Colors.ERROR;
    }

    return Colors.BLACK;
  }

  render() {
    const styles = StyleSheet.create({
      labelConainer: {
        flexDirection: 'row',
        width: '100%',
        marginTop: 8,
      },
      label: {
        fontSize: 12,
        color: this.labelState(),
      },
      input: {
        width: '100%',
        height: 48,
        borderWidth: 1,
        borderRadius: 4,
        marginBottom: 8,
        marginTop: 8,
        padding: 8,
        borderColor: this.borderState(),
      },
      icon: {
        justifyContent: 'center',
        alignItems: 'center',
        width: 40,
        height: 48,
        position: 'absolute',
        top: 8,
        right: 0,
      },
    });

    const { error, reference, form, index, label, toggle, info, password, infoTestID, infoAccessibilityLabel } = this.props;
    const { showPw } = this.state;
    const warning = error;
    const myProps = {
      style: styles.input,
    };
    let PwIcon = VSSvg.NoViewFillIcon;
    let LabelIcon = VSSvg.AlertFillIcon;
    let LabelIconColor = Colors.ERROR;
    let IconWrapper = null;

    if (!reference) {
      myProps.reference = (input) => {
        form[index] = input;
      };
    }

    if (info) {
      LabelIcon = VSSvg.InfoFillIcon;
      LabelIconColor = Colors.ORANGE;
      IconWrapper = ( // eslint-disable-line no-extra-parens
        <Touchable
          onPress={ toggle }
          testID={ infoTestID }
          accessibilityLabel={ infoAccessibilityLabel }
        >
          <LabelIcon color={ LabelIconColor } width={ 16 } height={ 16 } />
        </Touchable>
      );
    }

    if (warning) {
      LabelIcon = VSSvg.AlertFillIcon;
      LabelIconColor = Colors.ERROR;
      IconWrapper = <LabelIcon color={ LabelIconColor } width={ 16 } height={ 16 } />;
    }

    if (password) {
      if (showPw) {
        PwIcon = VSSvg.EyeFillIcon;
      }
      myProps.secureTextEntry = !showPw;
    }

    return (
      <View style={ styles.container }>
        <View style={ styles.labelConainer }>
          <Text style={ styles.label }>{ label }  </Text>
          { IconWrapper }
        </View>
        {
          password
            ? <View>
              <TextInput
                { ...myProps }
                { ...this.props }
                ref={ reference }
                underlineColorAndroid="rgba(0,0,0,0)"
                autoCorrect={ false }
              />
              <TouchableWithoutFeedback onPress={ this.togglePw }>
                <View style={ styles.icon } >
                  <PwIcon
                    color={ Colors.ORANGE }
                    width={ 16 }
                    height={ 16 }
                  />
                </View>
              </TouchableWithoutFeedback>
            </View>

            : <TextInput
              { ...myProps }
              { ...this.props }
              ref={ reference }
              underlineColorAndroid="rgba(0,0,0,0)"
              autoCorrect={ false }
            />
        }
      </View>
    );
  }
}

VSInput.propTypes = {
  label: PropTypes.string,
  form: PropTypes.array,
  error: PropTypes.string,
  index: PropTypes.number,
  info: PropTypes.bool,
  toggle: PropTypes.func,
  reference: PropTypes.func,
  password: PropTypes.bool,
  infoTestID: PropTypes.string,
  infoAccessibilityLabel: PropTypes.string,
};

VSInput.defaultProps = {
  form: [],
  index: 0,
};
