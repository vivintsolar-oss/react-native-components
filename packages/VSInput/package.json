{
  "name": "@vivintsolar-oss/native-vs-input",
  "version": "1.2.4",
  "description": "Input for VSLR UI kit",
  "private": false,
  "author": "tcasey",
  "main": "./index.js",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/vivintsolar-oss/react-native-components.git"
  },
  "license": "MIT",
  "dependencies": {
    "@vivintsolar-oss/native-vs-constants": "^1.0.2",
    "@vivintsolar-oss/native-vs-icons": "^1.2.3",
    "@vivintsolar-oss/native-vs-touchable": "^1.2.3"
  },
  "peerDependencies": {
    "prop-types": "^15.6.0",
    "react": "16.0.0",
    "react-native": "https://github.com/expo/react-native/archive/sdk-22.0.1.tar.gz"
  },
  "bugs": {
    "url": "https://github.com/vivintsolar-oss/react-native-components/issues"
  },
  "homepage": "https://github.com/vivintsolar-oss/react-native-components#readme",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "publishConfig": {
    "access": "public"
  }
}
