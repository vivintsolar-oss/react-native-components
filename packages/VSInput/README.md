# @vivintsolar-oss/native-vs-input

```
yarn add @vivintsolar-oss/native-vs-input
// or
// npm install @vivintsolar-oss/native-vs-input
```


## VSInput

Our input field adds some useful state changes along with keeping the look and feel consistent across multiple native apps.

``` javascript
import VSInput from '@vivintsolar-oss/native-vs-input';

  <VSInput
    password
    label="Password"
    reference={ (input) => {
      this.passwordInput = input;
    } }
    error={ this.state.error }
    autoCorrect={ false }
    onChangeText={ this.onPasswordChange }
    onSubmitEditing={ this.submit }
    returnKeyType="send"
    defaultValue={ this.state.password }
  />

```

## Props

| Name | Default | Type | Description |
|:----- |:------ |:---- |:----------- |
| label | label | string | The text for your label |
| reference |  | function | Ref function used to hoist state |
| info |  | boolean | Shows the info state |
| error |  | string | The error message displayed & shows error state |
| password |  | boolean | Shows the password state |
| toggle |  | function | Used to toggle state for more info icon (like a modal/alert) |

You can also add any other props available to the [react native TextInput component](https://facebook.github.io/react-native/docs/textinput.html#props)