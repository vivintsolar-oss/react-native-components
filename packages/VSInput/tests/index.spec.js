/* eslint-env node , jest */
import React from 'react';
import renderer from 'react-test-renderer';
import VSInput from '../';


describe('<VSInput />', () => {
  describe('snapshots', () => {
    describe('default props', () => {
      it('should match snapshot', () => {
        const rendered = renderer.create(
          <VSInput
            password
            label="Password"
            autoCorrect={ false }
            defaultValue={ '' }
          />
        ).toJSON();

        expect(rendered).toMatchSnapshot();
      });
    });
  });
});
