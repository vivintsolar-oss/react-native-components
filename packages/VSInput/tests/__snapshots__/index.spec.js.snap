// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VSInput /> snapshots default props should match snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 8,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#1F2123",
          "fontSize": 12,
        }
      }
    >
      Password
        
    </Text>
  </View>
  <View>
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      defaultValue=""
      form={Array []}
      index={0}
      label="Password"
      password={true}
      reference={[Function]}
      secureTextEntry={true}
      style={
        Object {
          "borderColor": "#D9DADB",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 48,
          "marginBottom": 8,
          "marginTop": 8,
          "padding": 8,
          "width": "100%",
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
    />
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 48,
          "justifyContent": "center",
          "position": "absolute",
          "right": 0,
          "top": 8,
          "width": 40,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            false,
            Object {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={23}
        vbWidth={25}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              1,
              0.5098039215686274,
              0,
              1,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          name="No_View_Fill"
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth="1"
          x={0}
          y={0}
        >
          <RNSVGPath
            d="M 23.1475 0L 18.7375 4.41C 17.0575 3.49001 14.9975 2.86 12.4975 2.86C 3.4675 2.86 0.0675049 11.09 0.0375061 11.17C -0.0124969 11.29 -0.0124969 11.42 0.0375061 11.54C 0.0574951 11.6 1.61749 15.33 5.3775 17.77L 1.14751 22L 1.8575 22.71L 23.8575 0.710007L 23.1475 0ZM 9.1075 11.35C 9.1075 9.42001 10.6775 7.85001 12.6075 7.85001C 13.3675 7.85001 14.0475 8.11 14.6175 8.52L 9.7775 13.36C 9.36749 12.79 9.1075 12.11 9.1075 11.35ZM 20.4475 5.5C 23.6175 7.91 24.9375 11.11 24.9575 11.16C 25.0075 11.28 25.0075 11.42 24.9475 11.55C 24.9175 11.63 21.5175 19.86 12.4875 19.86C 10.4375 19.86 8.6875 19.42 7.1875 18.76L 11.3375 14.61C 11.7275 14.76 12.1475 14.85 12.5975 14.85C 14.5275 14.85 16.0975 13.28 16.0975 11.35C 16.0975 10.91 16.0075 10.48 15.8575 10.09L 20.4475 5.5Z"
            fill={
              Array [
                0,
                0,
                0,
                0,
                1,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth="1"
            x={0}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
